ouroboros-consensus/docs/GenesisDesignOverview.md

```diff
@@ -0,0 +1,404 @@
+could be implemented as truncate them all before passing them as options to
+BlockFetch, except for blocks I meant (which currently circumvent BlockFetch).)
+
+```
+    Anchor_0 = intersection(S)
+
+Candidates_i = subset of S that extends Anchor_i
+
+     Bests_i = the (necessarily non-empty) subset of Candidates_i
+               whose density in the Genesis window starting at Anchor_i
+               could be the greatest
+               if some headers extending the element were to suddenly arrive
+
+Anchor_{i+1} = intersection(Bests_i)
+```
+
+The definition of `Bests_i` has some subtleties. In particular, if some element
+of `Bests_i` has a header that is at the end of or even beyond the Genesis
+window starting at `Anchor_i`, then that element has a fixed density in that
+window, since any new header extending the element would be outside of the
+Genesis window. If the element's tip occurs earlier within the window, then
+technically one extending header could arrive in each remaining slot in the
+window. So the maximum possible density for an element of `Bests_i` is
+conservatively defined as the number of its headers in the window plus the
+number of slots after its last header in the window.
+
+As our Syncing progress reaches the tip of the peers' chains, though, our peers
+will begin informing us that they have no additional headers. This scenario is
+converging on Praos's domain, since we actually have the peer's whole (header)
+chain itself, not just a prefix of it (TODO unless they're also syncing!). Thus
+we have the following refined definition.
+
+```
+    Cutoff_i = the minimum count of
+               headers in the Genesis window starting at Anchor_i
+               for any element of Candidates_i
+
+     Bests_i = the (necessarily non-empty) subset of Candidates_i
+               containing each prefix for which
+               the count of headers in the Genesis window starting at Anchor_i
+               plus the count of slots that are
+                   in the Genesis window starting at Anchor_i
+                   > the slot of the prefix's tip
+                   and <= the slot of the peer's claimed tip
+               is >= Cutoff_i
+
+TODO would it be clearer to map each prefix-and-claimed-tip pair to an interval
+[#headers in window, min(anchor+s, claimed tip) - #empty slots in window older
+than a header] and keep only those prefixes whose upper bound is >= the overall
+minimum lower bound?
+
+```
+
+Intuitively, an inferior candidate prefix is eliminated from consideration (and
+thereby stops holding back `Anchor_j`) in one of two ways. Either that chain
+ends in the window (while the better chains don't) or else the node eventually
+receives enough of each prefix in the window to see that the number of
+remaining slots in the window for this sparse prefix is too small for the
+prefix to ever be able to catch up to the greatest number of received headers
+in the window on whichever prefix has the most.
```

Comment:
is this how we ensure we've received "enough" headers?

I'm not clear on what the threshold is for prefix selection that makes it confident it's got enough headers to make a decision...